apps/snitch_core/lib/core/tools/validations.ex

```diff
@@ -0,0 +1,59 @@
+defmodule Snitch.Tools.Validations do
+  @moduledoc """
+  A bunch of data validations for `Ecto.Changeset`.
+
+  ## Note
+
+  All validations in this module are check ONLY the value present under the
+  `:changes` key in the `Ecto.Changeset.t()`.
+
+  The validations are non-strict, and will not complain if the key is not
+  present under `:changes`.
+  """
+
+  import Ecto.Changeset
+
+  @doc """
+  Validates that the amount (of type `Money.t`) under the `key` in `changeset`
+  is non-negative.
+  """
+  @spec validate_amount(Ecto.Changeset.t(), atom) :: Ecto.Changeset.t()
+  def validate_amount(%Ecto.Changeset{valid?: true} = changeset, key) when is_atom(key) do
```

Comment:
This can be removed as Ecto has same method for it `validate_number(changeset, field, opts)`
Ecto doc link :  [validate_number](https://hexdocs.pm/ecto/Ecto.Changeset.html#validate_number/3)
Check the defination : [validate_number](https://github.com/elixir-ecto/ecto/blob/v2.2.10/lib/ecto/changeset.ex#L1726)

Reply:
we can not use this one here, amount here is not a number but a `MoneyType`.